src/utils/init.js

```diff
@@ -61,28 +64,105 @@ function init(apiKey, options = {}) {
 
             if (!initState.userKey) initState.userKey = uuidv4();
             document.cookie = `feathery-user-id=${initState.userKey}; max-age=31536000`;
-            _fetchFormData(options.formKeys);
+            await _fetchFormData(options.formKeys);
         }
     }
 }
 
-// must be called after userKey loads
-function _fetchFormData(formKeys) {
-    formKeys.forEach((key) => {
-        const formClient = new Client(key);
-        formClient.fetchForm().then((stepsResponse) => {
-            initState.forms[key] = stepsResponse;
-        });
-        formClient.fetchSession().then((session) => {
-            initState.sessions[key] = session;
-        });
+function dynamicImport(dependency) {
+    return new Promise((resolve) => {
+        $script(dependency, resolve);
     });
 }
 
+const initializeIntegrations = async (
+    integrations,
+    clientArg,
+    init = false
+) => {
+    const gtm = integrations['google-tag-manager'];
+    if (gtm && !TagManager.initialized) {
+        TagManager.initialized = true;
+        TagManager.initialize({
+            gtmId: gtm.api_key,
+            dataLayer: { userId: initState.userKey }
+        });
+    }
+
+    const fb = integrations.firebase;
+    if (fb) {
+        let firebase = global.firebase;
+        if (!firebase) {
```

Comment:
`await` doesn't block execution outside its scope - I think you want to wait for this to finish before you continue on Line 112. You can wrap this all with a Promise:
```js
const firebase = await new Promise((resolve) => {
    if (!global.firebase) {
        await dynamicImport(...);
        ...
        resolve(global.firebase);
    } else {
        resolve(global.firebase);
    }
})